duration.go

```diff
@@ -122,33 +122,88 @@ func Parse(d string) (*Duration, error) {
 	return duration, nil
 }
 
+// Format formats the given duration into a ISO 8601 duration string (e.g. P1DT6H5M)
+func Format(d time.Duration) string {
+	neg := false
+	if d < 0 {
+		neg = true
+		d = -d
+	}
+
+	if d == 0 {
+		return "PT0S"
+	}
+
+	duration := &Duration{}
+	if d.Hours() >= 8760 {
+		duration.Years = math.Floor(d.Hours() / 8760)
+		d -= time.Duration(duration.Years) * time.Hour * 8760
+	}
+	if d.Hours() >= 730 {
+		duration.Months = math.Floor(d.Hours() / 730)
+		d -= time.Duration(duration.Months) * time.Hour * 730
+	}
+	if d.Hours() >= 168 {
+		duration.Weeks = math.Floor(d.Hours() / 168)
+		d -= time.Duration(duration.Weeks) * time.Hour * 168
+	}
+	if d.Hours() >= 24 {
+		duration.Days = math.Floor(d.Hours() / 24)
+		d -= time.Duration(duration.Days) * time.Hour * 24
+	}
+	if d.Hours() >= 1 {
+		duration.Hours = math.Floor(d.Hours())
+		d -= time.Duration(duration.Hours) * time.Hour
+	}
+	if d.Minutes() >= 1 {
+		duration.Minutes = math.Floor(d.Minutes())
+		d -= time.Duration(duration.Minutes) * time.Minute
+	}
+	duration.Seconds = d.Seconds()
+
+	if neg {
```

Comment:
ISO8601 durations cannot be negative. We shouldn't allow users to convert a `time.Duration` to an ISO8601 duration if it's negative. That should cause an error.

Reply:
I used [this spec](https://tc39.es/proposal-temporal/docs/duration.html#from) as a reference, and there is support for negative values

Reply:
Hmm. It does seem that the second extension of ISO8601 allows for a negative duration. I haven't bought the actual spec documents so I was working from the original version it seems.

Reply:
I'll close this question as resolved. Sorry about that.